src/main/java/agh/ics/oop/gui/GuiElementBox.java

```diff
@@ -0,0 +1,38 @@
+package agh.ics.oop.gui;
+
+import agh.ics.oop.IMapElement;
+import javafx.geometry.Pos;
+import javafx.scene.Node;
+import javafx.scene.control.Label;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.VBox;
+import javafx.scene.text.Font;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+
+public class GuiElementBox {
+    private final VBox vbox = new VBox();
+    private final ImageView imageView;
+
+    public GuiElementBox(IMapElement element) {
+        try {
+            FileInputStream file = new FileInputStream(element.toResource());
+            this.imageView = new ImageView(new Image(file));
+            this.imageView.setFitWidth(20);
+            this.imageView.setFitHeight(20);
+        } catch (FileNotFoundException ex) {
+            throw new RuntimeException(ex);
+        }
+    }
+
+    public Node toNode(String name) {
```

Comment:
to rozwiązanie sprawia, że część informacji jest cache'owana (obrazek), a część nie. Trochę dziwne i powinno powodować, że przy zmianie orientacji nie zmienia się obrazek. U Pana to działa, ale to dlatego, że... nie działa sam cache (patrz uwaga przy App).